crates/gpui/src/platform/test/platform.rs

```diff
@@ -360,7 +361,13 @@ impl Platform for TestPlatform {
     }
 
     fn set_menus(&self, _menus: Vec<crate::Menu>, _keymap: &Keymap) {}
-    fn set_dock_menu(&self, _menu: Vec<crate::MenuItem>, _keymap: &Keymap) {}
+    fn set_dock_menu(
+        &self,
+        _menu: Vec<crate::MenuItem>,
+        _keymap: &Keymap,
+    ) -> Vec<SmallVec<[PathBuf; 2]>> {
```

Comment:
I'd rather not change this API if we can. A `Vec<SmallVec<PathBuf>>` return type is a huge increase in complexity for what should be a pretty simple operation. If we have to have a `Vec<Vec<PathBuf>>`, I'd rather that be relegated to a windows-only API, than add it to the general purpose API.